tests/unit/service/conftest.py

```diff
@@ -620,10 +633,15 @@ async def feature_list_repeated_fixture(test_dir, feature, feature_list_service)
     )
     with open(fixture_path, encoding="utf") as fhandle:
         payload = json.loads(fhandle.read())
-        feature_list = await feature_list_service.create_document(
-            data=FeatureListServiceCreate(**payload)
-        )
-        return feature_list
+        feature_list = None
+        try:
+            feature_list = await feature_list_service.create_document(
+                data=FeatureListServiceCreate(**payload)
+            )
+            yield feature_list
+        finally:
+            if feature_list:
+                await feature_list_service.delete_document(document_id=feature_list.id)
```

Comment:
This `create_document` and `delete_document` pattern seems common enough that a helper context manager might be useful.

Reply:
Btw is it true that all tests that create a feature list will need to remember to clean up after use? Or only for specific types of tests?

Reply:
When the storage type is not function scope, would be good to clean it up when we create the feature list using fixed feature list ID (otherwise, between tests may have coupling effect). I encountered this a couple of times when fixing the tests (maybe before i added the suffix), found that it is useful to isolate the problem by just cleaning up the feature list.